.github/workflows/test.yml

```diff
@@ -0,0 +1,70 @@
+name: "Run tests"
+
+on:
+  pull_request:
+  push:
+    branches:
+      - develop
+      - master
+
+jobs:
+  test:
+    name: "Run unit tests"
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        php:
+          - '5.4'
+          - '5.5'
+          - '5.6'
+          - '7.0'
+          - '7.1'
+          - '7.2'
+          - '7.3'
+          - '7.4'
+          - '8.0'
+
+        include:
+          - php: '8.0'
+            sniffer: true
+
+    steps:
+      - name: Setup PHP
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: ${{ matrix.php }}
+          extensions: json, mbstring, bcmath
+          coverage: xdebug
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Get composer cache directory
```

Comment:
You may want to have a look at this action which does the running + caching of composer in one go: https://github.com/marketplace/actions/install-composer-dependencies